### PR TITLE
Add cubemap texture examples

### PR DESCRIPTION
**Pull request by dabingnn**

### CODE REVIEW

File: examples/src/texturecube.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      texture = new gfx.TextureCube(device, {
+        width : image.width, 
+        height : image.height,
+        images : [[assets.image0,assets.image1,assets.image2,assets.image3,assets.image4,assets.image5]]
+      });
+    }
+  });
+  let vertexFmt = new gfx.VertexFormat([
+    { name: gfx.ATTR_POSITION, type: gfx.ATTR_TYPE_FLOAT32, num: 3 },
+    { name: gfx.ATTR_COLOR, type: gfx.ATTR_TYPE_FLOAT32, num: 3 }
```
Comment (jwu) [inline examples/src/texturecube.js]: no need for color when program2 is removed.

Comment (dabingnn) [reply examples/src/texturecube.js]: Thanks, updated!

File: examples/src/texturecube.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
+(() => {
+  let gfx = window.gfx;
+  let device = window.device;
+  let canvas = window.canvas;
+  let resl = window.resl;
+  var texture = null;
```
Comment (jwu) [inline examples/src/texturecube.js]: var => let

File: examples/src/texturecube.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        width : image.width, 
+        height : image.height,
+        images : [[assets.image0,assets.image1,assets.image2,assets.image3,assets.image4,assets.image5]]
+      });
+    }
+  });
+  let vertexFmt = new gfx.VertexFormat([
+    { name: gfx.ATTR_POSITION, type: gfx.ATTR_TYPE_FLOAT32, num: 3 },
+  ]);
+  var verts = [
```
Comment (jwu) [inline examples/src/texturecube.js]: var => let

File: examples/src/texture.js [hunk 4]
```diff
@@ -56,7 +70,12 @@
     6,
     false
   );
-
+  let transform0 = mat4.create();
+  let transform1 = mat4.create();
+  let scale = vec3.create();
+  let trans = vec3.create();
+  mat4.fromRTS(transform0, quat.create(), vec3.set(trans, -0.5, 0, 0), vec3.set(scale, 0.5, 0.5, 0.5));
```
Comment (jwu) [inline examples/src/texture.js]: vec3.set => vec3.new, no need for scale and trans here

File: examples/src/texture.js [hunk 5]
```diff
@@ -28,6 +29,19 @@
     `,
   });
   program.link();
+  let canvasElement = document.createElement('Canvas');
```
Comment (jwu) [inline examples/src/texture.js]: Canvas => canvas

**Pull request merged**